contrib/kind-dualstack.md

```diff
@@ -0,0 +1,184 @@
+Commands above accepting local changes.
+
+## OVN-Kubernetes with IP dual-stack
+
+For status of IP dual-stack in OVN-Kubernetes, see
+[1142](https://github.com/ovn-org/ovn-kubernetes/issues/1142).
+
+To run OVN-Kubernetes with IP dual-stack in a KIND deployment, run:
+
+```
+$ go get github.com/ovn-org/ovn-kubernetes; cd $GOPATH/src/github.com/ovn-org/ovn-kubernetes
+
+$ cd go-controller/
+$ make
+
+$ cd ../dist/images/
+$ make fedora
+
+$ cd ../../contrib/
+$ IPV4_SUPPORT=true IPV6_SUPPORT=true K8S_VERSION=v1.18.0 ./kind.sh
+```
+
+Once `kind.sh` completes, setup kube config file:
+
+```
+$ cp ~/admin.conf ~/.kube/config
+-- OR --
+$ KUBECONFIG=~/admin.conf
+```
+
+Once testing is complete, to tear down the KIND deployment:
+
+```
+$ kind delete cluster --name ovn
+```
+
+## Current Status
+
+This is subject to change because code is being updated constantly. But this is
+more a cautionary note that this feature is not completely working at the
+moment.
+
+The nodes do not go to ready because the OVN-Kubernetes hasn't setup the network
+completely:
+
+```
+$ kubectl get nodes
+NAME                STATUS     ROLES    AGE   VERSION
+ovn-control-plane   NotReady   master   94s   v1.18.0
+ovn-worker          NotReady   <none>   61s   v1.18.0
+ovn-worker2         NotReady   <none>   62s   v1.18.0
+
+$ kubectl get pods -o wide --all-namespaces
+NAMESPACE          NAME                                      READY STATUS   RESTARTS AGE    IP          NODE
+kube-system        coredns-66bff467f8-hh4c9                  0/1   Pending  0        2m45s  <none>      <none>
+kube-system        coredns-66bff467f8-vwbcj                  0/1   Pending  0        2m45s  <none>      <none>
+kube-system        etcd-ovn-control-plane                    1/1   Running  0        2m56s  172.17.0.2  ovn-control-plane
+kube-system        kube-apiserver-ovn-control-plane          1/1   Running  0        2m56s  172.17.0.2  ovn-control-plane
+kube-system        kube-controller-manager-ovn-control-plane 1/1   Running  0        2m56s  172.17.0.2  ovn-control-plane
+kube-system        kube-scheduler-ovn-control-plane          1/1   Running  0        2m56s  172.17.0.2  ovn-control-plane
+local-path-storage local-path-provisioner-774f7f8fdb-msmd2   0/1   Pending  0        2m45s  <none>      <none>
+ovn-kubernetes     ovnkube-db-cf4cc89b7-8d4xq                2/2   Running  0        107s   172.17.0.2  ovn-control-plane
+ovn-kubernetes     ovnkube-master-87fb56d6d-7qmnb            3/3   Running  0        107s   172.17.0.2  ovn-control-plane
+ovn-kubernetes     ovnkube-node-278l9                        2/3   Running  0        107s   172.17.0.3  ovn-worker2
+ovn-kubernetes     ovnkube-node-bm7v6                        2/3   Running  0        107s   172.17.0.2  ovn-control-plane
+ovn-kubernetes     ovnkube-node-p4k4t                        2/3   Running  0        107s   172.17.0.4  ovn-worker
+```
+
+Current failure is setting up the Gateway:
```

Comment:
so do we want this included in this doc? Do you expect to update the current status here each time or @danwinship or whoever authors the fix to do it? Maybe just a comment in the PR on what the current state is would be better than putting it in the doc?

Reply:
True. I didn't plan on updating it with every change, but thought I revisit 2-3 times before dual stack was completed. Long term, not sure this doc is needed. It was more a short term guide to help with the development of dual stack.

Reply:
Left it as a more generic example where node doesn't go t oready and removed the specific error.